PIP_COMPATIBILITY.md

```diff
@@ -0,0 +1,317 @@
+
+## Local version identifiers
+
+uv does not implement spec-compliant handling of local version identifiers (e.g., `1.0.0+local`).
+Though local version identifiers are rare in published packages (and, e.g., disallowed on PyPI),
+they're common in the PyTorch ecosystem. uv's incorrect handling of local version identifiers
+may lead to resolution failures in some cases.
+
+In the future, uv will implement spec-compliant handling of local version identifiers.
+For more, see [#1855](https://github.com/astral-sh/uv/issues/1855).
+
+## Packages that exist on multiple indexes
+
+In both uv and `pip`, users can specify multiple package indexes from which to search for
+the available versions of a given package. However, uv and `pip` differ in how they handle
+packages that exist on multiple indexes.
+
+For example, imagine that a company publishes an internal version of `requests` on a private index
+(`--extra-index-url`), but also allow installing packages from PyPI by default. In this case, the
+private `requests` would conflict with the public [`requests`](https://pypi.org/project/requests/)
+on PyPI.
+
+When uv searches for a package across multiple indexes, it will iterate over the indexes in order
+(preferring the `--extra-index-url` over the default index), and stop searching as soon as it
+finds a match. This means that if a package exists on multiple indexes, uv will limit its
+candidate versions to those present in the first index that contains the package.
+
+(`pip`, meanwhile, will combine the candidate versions from all indexes, and select the best
+version from the combined set.)
+
+uv's behavior is such that if a package exists on an internal index, it should always be installed
+from the internal index, and never from PyPI. The intent is to prevent "dependency confusion"
+attacks, in which an attacker publishes a malicious package on PyPI with the same name as an
+internal package, thus causing the malicious package to be installed instead of the internal
+package. See, for example, [the `torchtriton` attack](https://pytorch.org/blog/compromised-nightly-dependency/)
+from December 2022.
+
+In the future, uv will support pinning packages to dedicated indexes (see: [#171](https://github.com/astral-sh/uv/issues/171)).
+Additionally, [PEP 708](https://peps.python.org/pep-0708/) is a provisional standard that aims to
+address the "dependency confusion" issue across package registries and installers.
+
+## Virtual environments by default
+
+`uv pip install` and `uv pip sync` are designed to work with virtual environments by default.
+
+Specifically, uv will always install packages into the currently active virtual environment, or
+search for a virtual environment named `.venv` in the current directory or any parent directory
+(even if it is not activated).
+
+This differs from `pip`, which will install packages into a global environment if no virtual
+environment is active, and will not search for inactive virtual environments.
```

Comment:
I'd add a word of warning here about the user and global env here, no isolation between different projects and different cli tools and the global env on linux can break system packages.